packages/dev/core/src/FlowGraph/Blocks/Data/flowGraphConditionalDataBlock.ts

```diff
@@ -0,0 +1,44 @@
+import type { FlowGraph } from "../../flowGraph";
+import { FlowGraphBlock } from "../../flowGraphBlock";
+import type { FlowGraphDataConnection } from "../../flowGraphDataConnection";
+
+/**
+ * @experimental
+ */
+export interface IFlowGraphConditionalDataBlockParams<T> {
+    defaultTrueValue: T;
+    defaultFalseValue: T;
+}
+
+/**
+ * @experimental
+ * Block that returns a value based on a condition.
+ */
+export class FlowGraphConditionalDataBlock<T> extends FlowGraphBlock {
+    public readonly condition: FlowGraphDataConnection<boolean>;
+    public readonly trueValue: FlowGraphDataConnection<T>;
+    public readonly falseValue: FlowGraphDataConnection<T>;
+
+    public readonly output: FlowGraphDataConnection<T>;
+
+    constructor(graph: FlowGraph, params: IFlowGraphConditionalDataBlockParams<T>) {

```

Comment:
same thing here - params could be an empty object, if its params are all optional